Documentation/applications/examples/xedge/index.rst

```diff
@@ -0,0 +1,131 @@
+``Xedge`` a lightweight Lua-based web framework for building secure, real-time IoT applications
+==============
+
+Xedge is a robust IoT and web framework that is designed for microcontrollers. It is based on the industrial-grade Barracuda Application Server, designed for seamless OEM integration. Xedge accelerates embedded firmware development by providing a flexible, Lua-based environment and a full stack of industrial-strength protocols, including:
+
+- OPC UA
+- Modbus
+- MQTT
+- SMQ
+- WebSockets
+- HTTP/HTTPS
+
+This Xedge port for NuttX comes pre-configured and requires:
+
+- TCP/IP v4 and v6 support
+- File System support
+- 2 MB RAM allocated statically in ``xedge/xedge_main.c``
+
+.. note::
+   These instructions set up Xedge in **development mode**. Xedge supports many configuration options that differ between development and production builds. For production settings and optimization, refer to the general Xedge build instructions (details below).
+
+Getting Started
+---------------
+
+To compile Xedge for NuttX, follow these steps:
+
+**Note:** the included script ``prepare.sh`` performs step 1 to 4.
+
+1. **Prepare the Xedge apps directory.**
+
+   Navigate to the Xedge application folder:
+
+   .. code-block:: bash
+
+      cd nuttxspace/apps/examples/xedge
+
+2. **Clone the required dependencies.**
+
+   .. code-block:: bash
+
+      git clone https://github.com/RealTimeLogic/BAS.git
+      git clone https://github.com/RealTimeLogic/BAS-Resources.git

```

Comment:
The Documentation is very good, but I think these steps don't need to be done by the users